photon-ml/src/main/scala/com/linkedin/photon/ml/model/Coefficients.scala

```diff
@@ -145,10 +147,13 @@ protected[ml] object Coefficients {
    */
   def apply(length: Int)(indices: Int*)(nnz: Double*): Coefficients = {
 
-    require(0 < length)
-    require(indices.length == nnz.length)
-    require(indices.sorted == indices)
-    // TODO: check for duplicates?
+    require(0 < length, s"Specified length < 0 for Coefficients: $length")
+    require(indices.length == nnz.length,
+      s"""There are ${indices.length} indices but ${nnz.length} non-zero specified for Coefficients.
+         | There should be as many indices as non-zero values.
+       """.stripMargin)
+    require(indices.sorted == indices, s"Indices need to be in strictly increasing order for Coefficients")
```

Comment:
Why not go ahead and sort the indices here? Also, this duplicates logic where we already enforce this invariant:

https://github.com/linkedin/photon-ml/blob/master/photon-ml/src/main/scala/com/linkedin/photon/ml/util/VectorUtils.scala#L55